### PR TITLE
produce rpm and deb file

### PR DESCRIPTION
**Pull request by edwardhartnett**
Part of #49 

In this PR I enable making .deb and .rpm packages.

@opoplawski is this helpful?

What is missing from the RPM for it to work for you?

(Build this branch and do `make package` to get the file.)

### CODE REVIEW

**Comment by kgerheiser** [comment]
Interesting. 

You get a failure on macOS if you include `DEB` and `RPM`.  Maybe a check if you're on Linux and then append to the list?

Otherwise I like it.

File: CMakeLists.txt [hunk 1]
```diff
@@ -42,6 +43,11 @@ include(CTest)
 if (BUILD_TESTING)
     add_subdirectory(tests)
 endif()
+
+# Create distributions.
+set(CPACK_PACKAGE_CONTACT "NOAA EMC")
+set(CPACK_GENERATOR "STGZ;TGZ;TZ;DEB;RPM")
```
Comment (kgerheiser) [inline CMakeLists.txt]: Generating `DEB` and `RPM` on non-Linux machines doesn't work

**Comment by opoplawski** [comment]
I'm looking to package this for Fedora, which requires its own spec and build.  So, no, this isn't helpful to me.  But it could be for many others.

**Comment by aerorahul** [comment]
can you also make homebrew packages for us lowly macOS users?

**Comment by edwardhartnett** [comment]
@aerorahul yes, we are turning on mac testing (see #53 ) and will then include macos...

File: CMakeLists.txt (hunk 1 above)
Comment (edwardhartnett) [reply CMakeLists.txt]: How do I specify in CMake that something should only  happen on Linux?

**Comment by edwardhartnett** [comment]
@kgerheiser can you approve please?

I am going to see if we can use rpm/yum to install this software...

**Review by kgerheiser — APPROVED** [review]

**Pull request merged**